docs/h5py_dataset.rst

```diff
@@ -299,6 +299,112 @@ what you requested, and nothing more.
 >>> print(data.shape)
 (80, 10)
 
+Variable-length data
+--------------------
+
+:class:`~.datasets.hdf5.H5PYDataset` also supports variable length data. Let's
+update the image features to reflect that:
+
+>>> train_image_features = [
+...     numpy.random.randint(10, size=(3, 5, 5)).astype('float32')
+...     for i in range(90)]
+>>> test_image_features = [
+...     numpy.random.randint(10, size=(3, 6, 6)).astype('float32')
+...     for i in range(10)]
+
+In this new example, training images measure 5x5 pixels whereas test images
+measure 6x6 pixels. Image sizes could also vary within the training and test
+sets; this particular arrangement is simply for illustrative purposes.
+
+First, we put the vector features and the targets inside the HDF5 file as
+before:
+
+>>> f = h5py.File('dataset.hdf5', mode='w')
+>>> f['vector_features'] = numpy.vstack(
+...     [numpy.load('train_vector_features.npy'),
+...      numpy.load('test_vector_features.npy')])
+>>> f['targets'] = numpy.vstack(
+...     [numpy.load('train_targets.npy'),
+...      numpy.load('test_targets.npy')])
+>>> f['vector_features'].dims[0].label = 'batch'
+>>> f['vector_features'].dims[1].label = 'feature'
+>>> f['targets'].dims[0].label = 'batch'
+>>> f['targets'].dims[1].label = 'index'
+
+We now have to put the variable-length images inside the HDF5 file. We can't
+do that directly, since HDF5 and h5py don't support multi-dimensional ragged
+arrays. However, there *is* support for 1D ragged arrays. Instead, we'll
+flatten the images before putting them in the HDF5 file:
+
+>>> all_image_features = train_image_features + test_image_features
+>>> dtype = h5py.special_dtype(vlen=numpy.dtype('uint8'))
+>>> image_features = f.create_dataset('image_features', (100,), dtype=dtype)
+>>> image_features[...] = [image.flatten() for image in all_image_features]
+>>> image_features.dims[0].label = 'batch'
+
+If you're feeling lost, have a look at the `dedicated tutorial on
+variable-length data`_.
+
+The images are now in the HDF5 file, but that doesn't help us unless we can
+recover their original shape. For that, we'll create a dimension scale that
+we'll attach to the ``'image_features'`` dataset using the name ``'shapes'``
+(use this *exact* name):
+
+>>> image_features_shapes = f.create_dataset(
+...     'image_features_shapes', (100, 3), dtype='uint8')
```

Comment:
If images are larger than 255 along one dimension, we would need a bigger dtype, I think. Is it worth mentioning as a note?